# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/WaylandBuffer.h
# Commit: 3927f21c92c9
# Full Hash: 3927f21c92c941c9efd0b00b0dc4cfc2b905452f
# Author: stransky <stransky@redhat.com>
# Date: 2025-09-08 08:22:14
# Description:
#   Bug 1979106 [Wayland] Don't recycle buffer transactions across WaylandSurfaces r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D263545
# ==============================================================================

diff -r 2a24e5601893 -r 3927f21c92c9 widget/gtk/WaylandBuffer.h
--- a/widget/gtk/WaylandBuffer.h	Mon Sep 08 02:52:34 2025 +0000
+++ b/widget/gtk/WaylandBuffer.h	Mon Sep 08 04:20:40 2025 +0000
@@ -64,7 +64,7 @@
 
   bool IsAttached() const;
 
-  BufferTransaction* GetTransaction();
+  BufferTransaction* GetTransaction(const WaylandSurfaceLock& aSurfaceLock);
   void RemoveTransaction(RefPtr<BufferTransaction> aTransaction);
 
 #ifdef MOZ_LOGGING
@@ -209,6 +209,10 @@
     return aBuffer == reinterpret_cast<uintptr_t>(mBuffer.get());
   }
 
+  bool CanRecycle(WaylandSurface* aSurface) {
+    return IsDetached() && (!mSurface || mSurface == aSurface);
+  }
+
  private:
   ~BufferTransaction();
 